src/Compilers/Core/Portable/PEWriter/Members.cs

```diff
@@ -184,7 +184,7 @@ ImmutableArray<byte> MarshallingDescriptor
         /// <summary>
         /// Offset of the field.
         /// </summary>
-        uint Offset
+        int Offset
         {
             get;
             // ^ requires this.ContainingTypeDefinition.Layout == LayoutKind.Explicit;
```

Comment:
Agreed. There is also some unnecessary redundancy on these interfaces that can be eliminated.